src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/FrameConnection.cs

```diff
@@ -49,7 +49,7 @@ public FrameConnection(FrameConnectionContext context)
         // Internal for testing
         internal PipeOptions AdaptedInputPipeOptions => new PipeOptions
         {
-            ReaderScheduler = InlineScheduler.Default,
+            ReaderScheduler = _context.ServiceContext.ThreadPool,
```

Comment:
Does this makes sense for the HTTPS adapter? Wouldn't that put unnecessary computation on the IO thread? The call to uv_write would be delayed/posted anyway.